content/develop/interact/search-and-query/advanced-concepts/vectors.md

```diff
@@ -9,369 +9,455 @@ categories:
-Higher values of `EF_RUNTIME` lead to more accurate results at the expense of a longer runtime. Default is 10.
-
-* `EPSILON` - Relative factor that sets the boundaries in which a range query may search for candidates. That is, vector candidates whose distance from the query vector is `radius*(1 + EPSILON)` are potentially scanned, allowing more extensive search and more accurate results (on the expense of runtime). Default is 0.01.   
+### Hash
+Store or update vectors and any metadata in [hashes]({{< baseurl >}}/develop/data-types/hashes/) with the [`HSET`]({{< baseurl >}}/commands/hset/) command.
 
 **Example**
-
 ```
-FT.CREATE my_index2 
-SCHEMA vector_field VECTOR 
-HNSW 
-16 
-TYPE FLOAT64 
-DIM 128 
-DISTANCE_METRIC L2 
-INITIAL_CAP 1000000 
-M 40 
-EF_CONSTRUCTION 250 
-EF_RUNTIME 20
-EPSILON 0.8
+HSET docs:01 vector <vector_bytes> foo bar
 ```
 
-## Indexing vectors
 
-### Storing vectors in hashes
-Storing vectors in Redis hashes is done by sending a binary blob of vector data. A common way of doing so is by using python numpy with redis-py client:
+{{% alert title="Tip" color="warning" %}}
+Hashes expect all data to be represented as strings. Thus, `<vector_bytes>` represents a bytes version of the vector.
+{{% /alert  %}}
+
+A common way of converting vectors to bytes is through the Python [numpy](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.tobytes.html) library and [redis-py](https://redis-py.readthedocs.io/en/stable/examples/search_vector_similarity_examples.html) client:
+
 ```py
 import numpy as np
 from redis import Redis
 
-redis_conn = Redis(host = 'localhost', port = 6379)
-vector_field = "vector"
-dim = 128
+redis_client = Redis(host='localhost', port=6379)
 
-# Store a blob of a random vector of type float32 under a field named 'vector' in Redis hash.
-np_vector = np.random.rand(dim).astype(np.float32)
-redis_conn.hset('key', mapping = {vector_field: np_vector.tobytes()})
-```
-Note that the vector blob size must match the vector field dimension and type specified in the schema, otherwise the indexing will fail in the background.  
+# Create a FLOAT32 vector
+vector = np.array([0.34, 0.63, -0.54, -0.69, 0.98, 0.61], dtype=np.float32)
 
-### Storing vectors in JSON
-Vector fields are supported upon indexing fields of JSON documents as well:
+# Convert vector to bytes
+vector_bytes = vector.tobytes()
 
-```
-FT.CREATE my_index ON JSON SCHEMA $.vec as vector VECTOR FLAT 6 TYPE FLOAT32 DIM 4 DISTANCE_METRIC L2 
+# Use redis client to store the vector bytes and metadata under a specified key
+redis_client.hset('docs:01', mapping = {"vector": vector_bytes, "foo": "bar"})
 ```
 
-Unlike in hashes, vectors are stored in JSON documents as arrays (not as blobs).
+{{% alert title="Tip" color="warning" %}}
+Note that the vector blob size must match the vector field dimension and float type specified in the schema, otherwise the indexing will fail in the background.
```

Comment:
`Note that the vector blob size must match the vector field dimension and float type specified in the schema, otherwise the indexing will fail`